frontend/SplitPane.js

```diff
@@ -39,30 +44,48 @@ class SplitPane extends React.Component {
     super(props);
     this.state = {
       moving: false,
-      width: props.initialWidth,
+      width: (props.isVertical) ? '100%' : props.initialWidth,
+      height: (!props.isVertical) ? '100%' : props.initialHeight,
     };
   }
 
-  onMove(x: number) {
+  componentWillReceiveProps(nextProps: Props) {
+    if (nextProps.isVertical !== this.props.isVertical) {
```

Comment:
The outer condition is now unnecessary because inner conditions always make it true.